WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/usecases/FetchScanStateUseCase.kt

```diff
@@ -27,32 +26,36 @@ class FetchScanStateUseCase @Inject constructor(
     suspend fun fetchScanState(
         site: SiteModel,
         polling: Boolean = true,
-        delayInMs: Long = FETCH_SCAN_STATE_DELAY_MILLIS
+        delayInMs: Long = FETCH_SCAN_STATE_DELAY_MILLIS,
+        startAfterDelay: Boolean = false
     ): Flow<FetchScanState> = flow {
-        if (!networkUtilsWrapper.isNetworkAvailable()) {
-            emit(Failure.NetworkUnavailable)
-            return@flow
-        }
+        while (true) {
+            if (!networkUtilsWrapper.isNetworkAvailable()) {
+                emit(Failure.NetworkUnavailable)
+                return@flow
+            }
 
-        val result = scanStore.fetchScanState(FetchScanStatePayload(site))
-        if (result.isError) {
-            emit(Failure.RemoteRequestFailure)
-            return@flow
-        } else {
-            val scanStateModel = scanStore.getScanStateForSite(site)
-            if (scanStateModel != null) {
-                emit(Success(scanStateModel))
-                if (polling && scanStateModel.state == ScanStateModel.State.SCANNING) {
-                    fetchScanState(site = site, polling = true).collect {
-                        emit(it)
+            if (startAfterDelay) {
+                delay(FETCH_SCAN_STATE_DELAY_MILLIS)
+            }
+
+            val result = scanStore.fetchScanState(FetchScanStatePayload(site))
+            if (result.isError) {
+                emit(Failure.RemoteRequestFailure)
+                return@flow
+            } else {
+                val scanStateModel = scanStore.getScanStateForSite(site)
+                if (scanStateModel != null) {
+                    emit(Success(scanStateModel))
+                    if (polling && scanStateModel.state == ScanStateModel.State.SCANNING) {
                         delay(delayInMs)
+                    } else {
+                        return@flow
                     }
                 } else {
+                    emit(Failure.RemoteRequestFailure)
                     return@flow
                 }
-            } else {
-                emit(Failure.RemoteRequestFailure)
-                return@flow
             }
         }
     }.flowOn(bgDispatcher)
```

Comment:
Praise (❤️): I really like that you put the `flowOn(...)` operator here. I know this is not part of this PR, but I wanted to comment that because I think all our use cases should switch to that approach. @zwarm , I am mentioning you here so that you are aware.

@malinajirka , wdyt?

PS: I believe this is the most robust way of using the use cases seems the client should not (or in those case, we don't want them) to manage the dispatcher themselves. The dispatcher needs to be a background dispatcher every time this use case is being used and as such it should be placed within the use case itself

Reply:
Passing on praise to @malinajirka, it was suggested by him in a PR review :).

Reply:
Praise to the lord, the almighty! 😄